syntax = "proto3";

import "common.proto";

service WorkerToScheduler {
    // Registers the worker with the scheduler
    rpc RegisterWorker (RegisterWorkerRequest) returns (RegisterWorkerResponse);

    // Indicates to the scheduler that a job has completed
    rpc Done (DoneRequest) returns (Empty);

    // Indicates to the scheduler that worker was killed
    rpc Killed (KilledRequest) returns (Empty);
}

message RegisterWorkerRequest {
    string device_id = 1;
    uint64 worker_id = 2;
    uint32 num_gpus = 3;
    string ip_addr = 4;
    uint32 port = 5;
}

message RegisterWorkerResponse {
    bool success = 1;
    uint64 worker_id = 2;
    string error_message = 3;
}

message DoneRequest {
    uint64 worker_id = 1;
}

message KilledRequest {
    uint64 worker_id = 1;
}